libs/ui/src/hooks/smartcard_auth/use_dipped_smartcard_auth.ts

```diff
@@ -79,16 +76,24 @@ function smartcardAuthReducer(
                 return { status: 'logged_out', reason: 'card_error' };
 
               case 'ready': {
-                const userResult = validateCardUser(action.card);
-                if (userResult.isOk()) {
-                  const user = userResult.ok();
+                const user = parseUserFromCard(action.card);
+                const validationResult = validateCardUser(user);
+                if (validationResult.isOk()) {
+                  assert(
+                    user &&
+                      (user.role === 'superadmin' || user.role === 'admin')
+                  );
```

Comment:
Is this assertion here because we only ever expect to use dipped smartcard auth for super admins and admins?